tools/topology/CMakeLists.txt

```diff
@@ -70,8 +70,8 @@ set(TPLGS
 	"sof-glk-da7219\;sof-glk-da7219"
 	"sof-glk-rt5682\;sof-glk-rt5682"
 	"sof-icl-nocodec\;sof-icl-nocodec"
-	"sof-apl-eq-pcm512x\;sof-apl-eq-pcm512x"
-	"sof-apl-eq-dmic\;sof-apl-eq-dmic"
+	"sof-apl-pcm512x-nohdmi\;sof-apl-eq-pcm512x\;-DEQ=enabled"
+	"sof-apl-dmic\;sof-apl-eq-dmic\;-DCHANNELS=4\;-DEQ=enabled"
```

Comment:
Can you make the -DEQ to some more generic options like `-DCOMP=EQ`, so you can also hare `-DCOMP=VOL` `-DCOMP=PASSTHROUGH`

Reply:
Better to have something like -DCAPTUREPROC=eq-iir-volume, -DPLAYBACKPROC=volume, etc. and tidy up all the processing pipelines so the applied processing pipeline are sof/pipe-CAPTUREPROC-capture.m4, sof/pipe-PLAYBACKPROC-playback.m4.

Reply:
> Can you make the -DEQ to some more generic options like `-DCOMP=EQ`, so you can also hare `-DCOMP=VOL` `-DCOMP=PASSTHROUGH`

Yep, we need also pipe-passthrough-playback.m4 and pipe-passthrough-capture.m4. Those should not contain any processing. Not even volume.

Also one thing I've been wondering is that what PCM name should the pipe-X-playback/capture.m4 define to be systematic? Now there's variation and it might cause extra challenge for user space audio servers. Or does it?

Reply:
@xiulipan @singalsu thank you for good comments. I read all your comments, i will work on making more general enhancement.

Reply:
Great, thanks @fredoh9 !